lib/SILGen/SILGenLValue.cpp

```diff
@@ -223,6 +223,63 @@ ManagedValue LogicalPathComponent::getMaterialized(SILGenFunction &SGF,
                                                    SILLocation loc,
                                                    ManagedValue base,
                                                    AccessKind kind) && {
+  if (getTypeOfRValue().getSwiftRValueType()->hasOpenedExistential()) {
```

Comment:
@jckarter @gottesmm This patch is really awkward. Can you think of a better way to address this problem?